docker-compose.yaml

```diff
@@ -160,5 +164,49 @@ services:
     ports:
       - 9090:9090
 
+  es:
+    image: docker.elastic.co/elasticsearch/elasticsearch:${STACK_VERSION}
+    container_name: es
+    hostname: "es.${CONTAINER_DOMAIN}"
+    networks:
+      - vault-playground
+    volumes:
+      - ./docker/es:/usr/share/elasticsearch/data:Z
+    ports:
+      - 9200:9200
+    environment:
+      - discovery.type=single-node
+      - xpack.security.enabled=false
+    mem_limit: ${MEM_LIMIT}
+
+  kibana:
+    image: docker.elastic.co/kibana/kibana:${STACK_VERSION}
+    container_name: kibana
+    hostname: "kibana.${CONTAINER_DOMAIN}"
+    networks:
+      - vault-playground
+    volumes:
+      - ./docker/kibana:/usr/share/kibana/data:Z
+    ports:
+      - 5601:5601
+    environment:
+      - ELASTICSEARCH_HOSTS=http://es.${CONTAINER_DOMAIN}:9200
+    mem_limit: ${MEM_LIMIT}
+
+  metricbeat:
+    image: docker.elastic.co/beats/metricbeat:${STACK_VERSION}
+    container_name: metricbeat
+    hostname: "metricbeat.${CONTAINER_DOMAIN}"
+    networks:
+      - vault-playground
+    volumes:
+      - ./docker/metricbeat/metricbeat.yml:/usr/share/metricbeat/metricbeat.yml
+      - ./docker/metricbeat/statsd.yml:/usr/share/metricbeat/modules.d/statsd.yml
+    ports:
+      - 8125:8125/udp
```

Comment:
- Metricbeat container listens on udp port 0.0.0.0:8125
- We map that port to the host system on port 8125 as well here
- The Vault server will send metrics to the IP of the Docker host (also port 8125)
- Metrics are then forwarded from Metricbeat to Elasticsearch